### PR TITLE
[COMCTL32][MSHTML][QCAP][MSGINA][UXTHEME][MSHTML_WINETEST][NTGDI][NTUSER] DeleteObject(hdc...) are wrong!

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Using `DeleteObject` against a DC is totally wrong. Use `DeleteDC` instead to delete a DC.

JIRA issue: N/A

## Proposed changes

- Use `DeleteDC` instead of `DeleteObject` against DCs.

## TODO

- [x] Build.
- [ ] Big tests.

### CODE REVIEW

File: win32ss/gdi/ntgdi/dclife.c [hunk 1]
```diff
@@ -930,7 +930,7 @@ IntGdiDeleteDC(HDC hDC, BOOL Force)
 
     if (GreIsHandleValid(hDC))
     {
-        if (!GreDeleteObject(hDC))
+        if (!IntGdiDeleteDC(hDC, FALSE))
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/dclife.c:933]: hm you're doing a recursive call there.

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: win32ss/gdi/ntgdi/dclife.c (hunk 1 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/dclife.c]: Reverted.

**Comment by tkreuzer** [comment]
No, it's not "totally wrong". While it's not documented, you can call DeleteObject with a DC just fine on Windows. A quick look at our implementation of DeleteObject also reveals that.
I'm fine with changing this in our own code, but not in wine, since it fixes nothing and only causes useless deviation from the original code.

**Review by tkreuzer — CHANGES_REQUESTED** [review]
Don't change wine code, since the "fix" doesn't actually fix anything.

**Comment by katahiromz** [comment]
Sorry my half-rotten brain. My brain is weak in undocumented things. Closed.

**Pull request closed without merging**